src/main/java/org/openmrs/addonindex/backend/SupportsDownloadCounts.java

```diff
@@ -0,0 +1,22 @@
+package org.openmrs.addonindex.backend;
+
+import org.openmrs.addonindex.domain.AddOnInfoAndVersions;
+import org.openmrs.addonindex.domain.AddOnToIndex;
+
+/**
+ *Represents a Strategy for fetching number of downloads over the past 30 days.
```

Comment:
Actually, in this comment and in the method-level comment, please don't be so specific about the "in last 30 days". The idea would be that various back-ends might allow us to fetch different kinds of download counts, and we'll have to be flexible about it.
So that means that when we _store_ a download count we need to be precise about what kind (which is why I said it's important to say "in last 30 days" in the field name that we're storing), but the code should expect that we'll eventually need to support other things, hence this method should just say "download counts"